server/grpc_service.go

```diff
@@ -1621,7 +1621,12 @@ func (s *GrpcServer) ScatterRegion(ctx context.Context, request *pdpb.ScatterReg
 		return nil, err
 	}
 	if op != nil {
-		rc.GetOperatorController().AddOperator(op)
+		if !rc.GetOperatorController().AddOperator(op) {
```

Comment:
Maybe this could be merged into one `if` block like:

```suggestion
		if op != nil && !rc.GetOperatorController().AddOperator(op) {
```